include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -7176,11 +7176,24 @@ namespace VectorTools
               Assert (n_components >= dim,
                       ExcMessage ("You can only ask for the Hdiv norm for a finite element "
                                   "with at least 'dim' components. In that case, this function "
-                                  "will take the divergence of the first 'dim' components."));
+                                  "will find the index of the first non-zero weight and take "
+                                  "the divergence of the 'dim' components that follow it."));
+
+              unsigned int idx = 0;
+              if (weight!=nullptr)
+                {
+                  Vector<double> tmp(n_components);
+                  weight->vector_value(Point<spacedim>(), tmp);
```

Comment:
It seems wasteful to allocate memory just to check whether or not we have a nonzero component. Could you just add this `Vector` to `IDScratchData` to avoid this?

Reply:
I will change this, I was thinking of using the already existing `weight_vectors` for this, since it does not really matter for this purpose at which point the `ComponentSelectFunction` is evaluated.